src/test/java/org/openrewrite/kotlin/tree/AnnotationTest.java

```diff
@@ -82,4 +84,123 @@ void jvmNameAnnotation() {
           )
         );
     }
+
+    @Test
+    void fullyQualifiedCustomAnnotation() {
+        rewriteRun(
+          java(
+            """
+              package com.example.annotations;
+                            
+              import java.lang.annotation.Target;
+              import static java.lang.annotation.ElementType.*;
+                            
+              @Target({TYPE, FIELD, METHOD, PARAMETER, CONSTRUCTOR, LOCAL_VARIABLE})
+              public @interface JavaAnnotationToRemove {
+              }
+              """
+          ),
+          // Works
+          kotlin(
+            """
+              import com.example.annotations.JavaAnnotationToRemove
+
+              @JavaAnnotationToRemove
+              class A {
+              }
+              """),
+          // Fails, despite being valid Kotlin code
+          kotlin(
+            """
+              @com.example.annotations.JavaAnnotationToRemove
+              class A {
+              }
+              """)
+        );
+    }
+
+    @Test
+    void fullyQualifieJavaLangAnnotation() {
+        rewriteRun(
+          // Works, Deprecated is defined in java.lang package so no import needed
+          kotlin(
+            """
+              @Deprecated
+              class A {
+              }
+              """),
+          // Fails, despite being valid Kotlin code
+          kotlin(
+            """
+              @java.lang.Deprecated
+              class A {
+              }
+              """)
+        );
+    }
+
+    // Despite being recognized as an annotation, it is not removed (but it works fine with annotations from java.lang package)
```

Comment:
Appreciate the clarification here while we work through these failures; typically we put that context in the issue before we merge, and then have the test method name reflect what we (don't) want to happen.